openshift_cli_installer/libs/clusters/ipi_cluster.py

```diff
@@ -181,3 +182,75 @@ def destroy_cluster(self):
         self.run_installer_command(action=DESTROY_STR, raise_on_failure=True)
         self.logger.success(f"{self.log_prefix}: Cluster destroyed")
         self.delete_cluster_s3_buckets()
+
+
+class AwsIpiCluster(IpiCluster):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.logger = get_logger(f"{self.__class__.__module__}-{self.__class__.__name__}")
+
+        self.platform = AWS_STR
+        if self.create:
+            self._create_install_config_file(platform=self.platform)
+        self.dump_cluster_data_to_file()
+
+
+class GcpIpiCluster(IpiCluster):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.logger = get_logger(f"{self.__class__.__module__}-{self.__class__.__name__}")
+
+        self.platform = GCP_STR
+        self.gcp_sa_file_dir = os.path.join(os.path.expanduser("~"), ".gcp")
+        self.gcp_sa_file_path = os.path.join(self.gcp_sa_file_dir, "osServiceAccount.json")
+        self.tmp_path = os.path.join("/tmp", "installerTemp.json")
+
+        if self.create:
+            self.terraform_parameters.update({"gcp_project_id": self.get_gcp_project_id()})
+            self._create_install_config_file(platform=self.platform)
+
+        self.dump_cluster_data_to_file()
+
+    def get_gcp_project_id(self):
+        with open(self.gcp_service_account_file, "r") as fd:
+            gcp_sa_file_content = json.load(fd)
+
+        return gcp_sa_file_content["project_id"]
+
+    def set_gcp_service_account_file(self):
+        """
+        Saves GCP Service Account file at location '~/.gcp/osServiceAccount.json'
+        If there is already file exists at this path, it will be saved at tmp path first.
+
+        """
+        if os.path.exists(self.gcp_sa_file_path) and not os.path.exists(self.tmp_path):
+            self.logger.info(f"File {self.gcp_sa_file_path} already exists. Copying to {self.tmp_path}")
+            shutil.copy(self.gcp_sa_file_path, self.tmp_path)
+        else:
+            Path(self.gcp_sa_file_dir).mkdir(parents=True, exist_ok=True)
+        self.logger.info(f"Saving GCP ServiceAccount file to {self.gcp_sa_file_path}")
+        shutil.copy(self.gcp_service_account_file, self.gcp_sa_file_path)
+
+    def unset_gcp_service_account_file(self):
+        """
+        Removes directory '~/.gcp' or Restores file at location
+         '~/.gcp/osServiceAccount.json' if exists before.
+
+        """
+        if os.path.exists(self.tmp_path):
```

Comment:
@rnetser sonarqube check was failing for using `/tmp` location directly, so updated the code to use tempfile module instead for creating backup_file, and it will create file with unique name, so had to remove shortuuid from it.